homeassistant/components/aprilaire/humidifier.py

```diff
@@ -0,0 +1,194 @@
+"""The Aprilaire humidifier component."""
+
+from __future__ import annotations
+
+from collections.abc import Awaitable, Callable
+from dataclasses import dataclass
+from typing import Any, cast
+
+from pyaprilaire.const import Attribute
+
+from homeassistant.components.humidifier import (
+    HumidifierAction,
+    HumidifierDeviceClass,
+    HumidifierEntity,
+    HumidifierEntityDescription,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.typing import StateType
+
+from .const import DOMAIN
+from .coordinator import AprilaireCoordinator
+from .entity import BaseAprilaireEntity
+
+HUMIDIFIER_ACTION_MAP: dict[StateType, HumidifierAction] = {
+    0: HumidifierAction.IDLE,
+    1: HumidifierAction.IDLE,
+    2: HumidifierAction.HUMIDIFYING,
+    3: HumidifierAction.OFF,
+}
+
+DEHUMIDIFIER_ACTION_MAP: dict[StateType, HumidifierAction] = {
+    0: HumidifierAction.IDLE,
+    1: HumidifierAction.IDLE,
+    2: HumidifierAction.DRYING,
+    3: HumidifierAction.DRYING,
+    4: HumidifierAction.OFF,
+}
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up Aprilaire humidifier devices."""
+
+    coordinator: AprilaireCoordinator = hass.data[DOMAIN][config_entry.unique_id]
+
+    assert config_entry.unique_id is not None
+
+    descriptions: list[AprilaireHumidifierDescription] = []
+
+    if coordinator.data.get(Attribute.HUMIDIFICATION_AVAILABLE) in (0, 1, 2):
```

Comment:
Just a nitpick for later, maybe we could avoid those magic numbers and use some speaking consts for it :)

Reply:
Ya definitely a fair point. These are unfortunately somewhat magic numbers in the documentation itself, but perhaps I could think of a way to better describe them.